cassandra-mesos-model/src/main/java/io/mesosphere/mesos/util/CassandraFrameworkProtosUtils.java

```diff
@@ -339,7 +339,7 @@ public ResourceHasExpectedRole(@NotNull final String role) {
 
         @Override
         public boolean apply(final Resource item) {
-            return item.getRole().equals(role);
+            return "*".equals(role) || "*".equals(item.getRole()) || item.getRole().equals(role);
```

Comment:
For simplicity this should be an all or nothing match.

Either all resources will be available in the offer matching the specified role, or not.

Also I think `"*".equals(role)` could cause true to be returned in a scenario where the resource didn't actually have the `*` role.